app/Controllers/importExportController.php

```diff
@@ -309,7 +309,7 @@ private function importJson(string $article_file, bool $starred = false): bool {
 		$limits = FreshRSS_Context::$system_conf->limits;
 
 		// First, we check feeds of articles are in DB (and add them if needed).
-		foreach ($items as $item) {
+		foreach ($items as &$item) {
```

Comment:
What's the specific scenario that can lead to this being necessary?

Reply:
This is relevant for all the cases when we make changes to `item`, otherwise our changes are not persisted.

```php
<?php

$items = [
	'a' => ['a' => 'b'],
	'b' => ['a' => 'b'],
];

foreach ($items as $item) {
	$item['a'] = 'c';
}

echo json_encode($items);
// Prints: {"a":{"a":"b"},"b":{"a":"b"}}

foreach ($items as &$item) {
	$item['a'] = 'c';
}

echo json_encode($items);
// Prints: {"a":{"a":"c"},"b":{"a":"c"}}
```

Reply:
I just figured that was generally a negative (within the code here), is all. :-)